pkg/util/tail/tail_test.go

```diff
@@ -18,10 +18,92 @@ package tail
 
 import (
 	"bytes"
+	"io/ioutil"
+	"k8s.io/apimachinery/pkg/util/diff"
+	"os"
+	"reflect"
 	"strings"
 	"testing"
 )
 
+type ReadAtMostReturn struct {
+	data                []byte
+	IsOffsetGreaterZero bool
+	err                 error
+}
+
+func TestReadAtMost(t *testing.T) {
+	//a file "test data A "
+	//b file "test data B"
+	//Append file B to the back of file A
+	//There will be three results, the file size is enough, the file size is just right, the file size is not enough
+	//Return the expected size data form the end of the file
+
+	//Create the tmp file and write data
+	fileDir, err := ioutil.TempDir("", "file-")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(fileDir)
+	fileOpen, err := os.OpenFile(fileDir+"/testFileA", os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0666)
+	if err != nil {
+		t.Fatalf("Failed to create file %s: %v", fileDir+"/testFileA", err)
+	}
+	defer fileOpen.Close()
+
+	if _, err = fileOpen.WriteString("test data A "); err != nil {
```

Comment:
Can you just write the complete string via ioutil.WriteFile? I don't understand why you would do it in two steps like this.

Reply:
If it's to get the length, you can just declare two constants and call len() on them separately, no?

Reply:
ok, thanks